[docs] Explain how to manage focus with `renderCell` (#4254)
mui/mui-x: docs/data/data-grid/accessibility/accessibility.md

The grid responds to keyboard interactions from the user and emits events when key presses happen on the grid cells.

### Tab sequence

According to [WCAG](https://www.w3.org/TR/wai-aria-practices-1.1/#grid), only one of the focusable elements contained by the grid should be included in the page tab sequence.

> For an element to be included in the tab sequence it needs to have a `tab-index` value of zero or greater.

Reviewer: We can treat it as a prop.

```suggestion
For an element to be included in the tab sequence, it needs to have a `tabIndex` value of zero or greater.
```

> When a cell of the data grid is focused, the first element with the attribute `tab-index=0` will receive the focus.

Reviewer: Not sure if "inner element" makes sense but I want to say that is the first element INSIDE the focused cell.

```suggestion
When a cell of the grid is focused, the first inner element with `tabIndex=0` will receive the focus.
```

> If there is no element with `tab-index=0` the focus is set on the cell itself.

Reviewer: ```suggestion
If there is no element with `tabIndex=0`, the focus is set on the cell itself.
```

> In the example below, the second grid does not remove links from the tab sequence, which implies having to pass through all the links before accessing the pagination controls.

Reviewer: Make it the first? So we first show the wrong and them the correct approach.

```suggestion
In the example below, the first grid does not remove links from the tab sequence, which implies having to pass through all the links before accessing the pagination controls.
```

This behavior makes it complicated to navigate between elements when using large datasets.

{{"demo": "FocusManagement.js", "bg": "inline", "defaultCodeOpen": false}}

> If you are customizing cell rendering with [`renderCell`](/x/react-data-grid/columns/#render-cell) method, you become responsible for removing focusable elements from the page tab sequence.

Reviewer: ```suggestion
If you are customizing cell rendering with the [`renderCell`](/x/react-data-grid/columns/#render-cell) method, you become responsible for removing focusable elements from the page tab sequence.
```

> To do so, use the `tabIndex` prop passed to the `renderCell` params, to know if the rendered cell has the focus, and so if the elements should be removed from the tab sequence.

Reviewer: ```suggestion
To do so, use the `tabIndex` prop passed to the `renderCell` params to know if the rendered cell has focus and if the element should be removed from the tab sequence.
```
